agent/consul/leader_connect_test.go

```diff
@@ -450,21 +440,26 @@ func TestCAManager_RenewIntermediate_Vault_Primary(t *testing.T) {
 	verifyLeafCert(t, caRoot, cert.CertPEM)
 }
 
+func patchIntermediateCertRenewInterval(t *testing.T) {
+	origInterval := structs.IntermediateCertRenewInterval
+	origMinTTL := structs.MinLeafCertTTL
+
+	structs.IntermediateCertRenewInterval = 200 * time.Millisecond
```

Comment:
I think this is one of the two fixes that should resolve the flakes. Previously we only waited 1ms, which I think meant that in some cases the renew goroutine could run before we got a reference to the "original" certificate.  The test does not run any slower with this 200ms. I believe it actually runs faster than it does on `main` because of other changes in this PR.